### ISSUE TITLE
Why do we need to evaluate the actor_critic model twice?

### ISSUE THREAD

**Issue by maximilianigl** [ikostrikov/pytorch-a2c-ppo-acktr#42]
I'm sorry if I'm missing something obvious.

What is the reason we have to evaluate the model twice, i.e. once when we call `actor_critic.act(...)` 
([here](https://github.com/ikostrikov/pytorch-a2c-ppo-acktr/blob/66eb0fa5d45c3b7da2e9a115525d61bb9ad4b111/main.py#L120)) and a second time  when we call `actor_critic.evaluate(...)` ([here](https://github.com/ikostrikov/pytorch-a2c-ppo-acktr/blob/66eb0fa5d45c3b7da2e9a115525d61bb9ad4b111/main.py#L154)). 

(I so far have only looked at A2C)

I understand that the computation graph isn't saved in the current implementation from when we call `act`. But couldn't we just save `values`,`action_log_probs` and `dist_entropy` in a list of `Variable`s when we call `act`?

Thanks for your help (and the implementation)!

**Comment by ikostrikov** [comment]
See the beginning of the discussion here: https://github.com/ikostrikov/pytorch-a2c-ppo-acktr/issues/15

tl;dr: It's just more efficient to do it this way on GPU since we can use batched computations for the backward pass.

**Comment by maximilianigl** [comment]
Ah great, thanks for the fast reply and sorry I missed that!
